packages/openid4vc-client/tests/openid4vc-client.e2e.test.ts

```diff
@@ -98,99 +109,57 @@ describe('OpenId4VcClient', () => {
-        Location: 'https://launchpad.vii.electron.mattrlabs.io/.well-known/openid-credential-issuer',
-      })
-
-      // setup server metadata response
-      const httpMock = nock('https://launchpad.vii.electron.mattrlabs.io')
-        .get('/.well-known/openid-credential-issuer')
-        .reply(200, getMetadataResponse)
-
-      // setup access token response
-      httpMock.post('/oidc/v1/auth/token').reply(200, acquireAccessTokenResponse)
-
-      // setup credential request response
-      httpMock.post('/oidc/v1/auth/credential').reply(200, credentialRequestResponse)
-    })
 
-    afterAll(async () => {
-      cleanAll()
-      enableNetConnect()
-    })
+  xdescribe('Authorization flow', () => {
+    const openidCredentialOffer = AUTHORIZED_OPENID_CREDENTIAL_OFFER
 
     it('should generate a valid authorization url', async () => {
       const clientId = 'test-client'
 
       const redirectUri = 'https://example.com/cb'
       const scope = ['TestCredential']
-      const initiationUri =
-        'openid-initiate-issuance://?issuer=https://launchpad.mattrlabs.com&credential_type=OpenBadgeCredential'
+
       const { authorizationUrl } = await agent.modules.openId4VcClient.generateAuthorizationUrl({
         clientId,
         redirectUri,
         scope,
-        initiationUri,
+        openidCredentialOffer,
       })
 
-      const parsedUrl = new URL(authorizationUrl)
-      expect(authorizationUrl.startsWith('https://launchpad.vii.electron.mattrlabs.io/oidc/v1/auth/authorize')).toBe(
-        true
-      )
-      expect(parsedUrl.searchParams.get('response_type')).toBe('code')
-      expect(parsedUrl.searchParams.get('client_id')).toBe(clientId)
-      expect(parsedUrl.searchParams.get('code_challenge_method')).toBe('S256')
-      expect(parsedUrl.searchParams.get('redirect_uri')).toBe(redirectUri)
-    })
-    it('should throw if no scope is provided', async () => {
-      // setup temporary redirect mock
-      nock('https://launchpad.mattrlabs.com').get('/.well-known/openid-credential-issuer').reply(307, undefined, {
-        Location: 'https://launchpad.vii.electron.mattrlabs.io/.well-known/openid-credential-issuer',
-      })
+      const url = new URL(authorizationUrl)
+      const params = url.searchParams
 
-      // setup server metadata response
-      nock('https://launchpad.vii.electron.mattrlabs.io')
-        .get('/.well-known/openid-credential-issuer')
-        .reply(200, getMetadataResponse)
+      expect(url.origin).toStrictEqual('https://launchpad.vii.electron.mattrlabs.io')
+      expect(url.pathname).toStrictEqual('/oidc/v1/auth/authorize')
 
-      const clientId = 'test-client'
-      const redirectUri = 'https://example.com/cb'
-      const initiationUri =
-        'openid-initiate-issuance://?issuer=https://launchpad.mattrlabs.com&credential_type=OpenBadgeCredential'
+      expect(params.get('response_type')).toBe('code')
```

Comment:
This errors now because the params of the authorizationUrl are incorrect.

Right now the entire JSON body is the key and the value is an empty string. It should be fixed by making every key value in the JSON object a separate query parameter key-value pair.

Reply:
should be fixed in latest feature/issuer commit